central/declarativeconfig/manager_impl.go

```diff
@@ -215,8 +246,127 @@ func (m *managerImpl) reconcileTransformedMessages(transformedMessagesByHandler
+				// Ensure the error count hasn't updated in the mean time.
+				if m.errorsPerDeclarativeConfig[messageID] == currentDeclarativeConfigErrors {
+					m.errorsPerDeclarativeConfig[messageID] = 0
+				}
+			})
+		}
+
+		m.declarativeConfigErrorReporter.UpdateIntegrationHealthAsync(&storage.IntegrationHealth{
+			Id:            messageID,
+			Name:          stringutils.FirstNonEmpty(extractNameFromProtoMessage(message), messageID),
+			Type:          storage.IntegrationHealth_DECLARATIVE_CONFIG,
+			Status:        storage.IntegrationHealth_HEALTHY,
+			ErrorMessage:  "",
+			LastTimestamp: timestamp.TimestampNow(),
+		})
+	}
+
+	log.Debugf("Error within reconciliation for %+v: %v", message, err)
+}
+
+// updateHandlerHealth will update the health status of a handler using the integrationhealth.Reporter.
+func (m *managerImpl) updateHandlerHealth(handlerID string, err error) {
+	var errMsg string
+	if err != nil {
+		errMsg = err.Error()
+	}
+
+	m.declarativeConfigErrorReporter.UpdateIntegrationHealthAsync(&storage.IntegrationHealth{
+		Id:            handlerID,
+		Name:          fmt.Sprintf("Config Map %s", path.Base(handlerID)),
+		Type:          storage.IntegrationHealth_DECLARATIVE_CONFIG,
+		Status:        utils.IfThenElse(err != nil, storage.IntegrationHealth_UNHEALTHY, storage.IntegrationHealth_HEALTHY),
+		ErrorMessage:  errMsg,
+		LastTimestamp: timestamp.TimestampNow(),
+	})
+}
+
+func (m *managerImpl) registerHealthForMessage(messages ...proto.Message) {
+	for _, message := range messages {
+		messageID := extractIDFromProtoMessage(message)
+		messageName := extractNameFromProtoMessage(message)
+
+		if err := m.declarativeConfigErrorReporter.Register(messageID, stringutils.FirstNonEmpty(messageName, messageID),
+			storage.IntegrationHealth_DECLARATIVE_CONFIG); err != nil {
+			log.Errorf("Error registering health status for declarative config %+v: %v", message, err)
+		}
+	}
+}
+
+// Helpers.
+
+func extractIDFromProtoMessage(message proto.Message) string {
+	// Special case, as the group specifies the ID nested within the groups properties.
+	if group, ok := message.(*storage.Group); ok {
+		return group.GetProps().GetId()
+	}
+	// Special case, as the name of the role is the ID.
+	if role, ok := message.(*storage.Role); ok {
+		return role.GetName()
+	}
+
+	messageWithID, ok := message.(interface {
+		GetId() string
+	})
+	// Theoretically, this should never happen unless we add more proto messages to the reconciliation. Hence, we use
+	// utils.Should to guard this.
+	if !ok {
+		utils.Should(errox.InvariantViolation.Newf("could not retrieve ID from message type %T %+v",
+			message, message))
+		return ""
+	}
+	return messageWithID.GetId()
+}
+
+func extractNameFromProtoMessage(message proto.Message) string {
+	messageWithName, ok := message.(interface {
+		GetName() string
+	})
+	// This may happen for some resources (such as groups, roles) as they do not define a name.
+	if !ok {
```

Comment:
funny thing is that within the YAML format, we introduce names for groups, the problem is we lose this data during the transformation to proto

Reply:
Yeah its not possible to retrieve the name from the UUID, so I guess for groups we have to live with this sadly.
Alternatively, we could use as name the tuple role, auth provider id, key, value, but that also doesn't make things easier.